enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/statemachine/guard/document/RequiredDocumentTypesGuard.java

```diff
@@ -44,31 +46,58 @@ public class RequiredDocumentTypesGuard {
+                .collect(toList());
+
+        if (!isTwoDistinctDocumentsPresent(acceptedDocumentVerifications)) {
+            logger.debug("There is not enough accepted document yet for identity verification ID: {}", identityVerificationId);
             return false;
-        } else if (!containsIdOrPassport(documentVerifications)) {
-            logger.debug("There is no accepted ID card or travel passport yet for identity verification ID: {}", identityVerificationId);
+        } else if (!containsPrimaryDocument(acceptedDocumentVerifications)) {
+            logger.debug("There is no accepted primary document yet for identity verification ID: {}", identityVerificationId);
             return false;
-        } else if (!containsDrivingLicence(documentVerifications)) {
-            logger.debug("There is no accepted driving licence yet for identity verification ID: {}", identityVerificationId);
+        } else if (!containsSecondaryDocument(acceptedDocumentVerifications)) {
+            logger.debug("There is no accepted secondary document yet for identity verification ID: {}", identityVerificationId);
             return false;
         } else {
             logger.debug("All required documents accepted for identity verification ID: {}", identityVerificationId);
             return true;
         }
     }
 
-    private static boolean containsIdOrPassport(final Collection<DocumentVerificationEntity> documentVerifications) {
+    private static boolean isTwoDistinctDocumentsPresent(final Collection<DocumentVerificationEntity> documentVerifications) {
+        return 2 == documentVerifications.stream()
+                .map(DocumentVerificationEntity::getType)
+                .distinct()
+                .count();
+    }
+
+    private static boolean containsPrimaryDocument(final Collection<DocumentVerificationEntity> documentVerifications) {
+        return containsBothSidesOfId(documentVerifications) || containsPassport(documentVerifications);
+    }
+
+    private static boolean containsBothSidesOfId(final Collection<DocumentVerificationEntity> documentVerifications) {
+        return 2 == documentVerifications.stream()
+                .filter(it -> it.getType() == DocumentType.ID_CARD)
+                .map(DocumentVerificationEntity::getSide)
+                .distinct()
+                .count();
+    }
+
+    private static boolean containsPassport(final Collection<DocumentVerificationEntity> documentVerifications) {
         return documentVerifications.stream()
-                .filter(it -> it.getStatus() == DocumentStatus.ACCEPTED)
                 .map(DocumentVerificationEntity::getType)
-                .anyMatch(it -> it == DocumentType.ID_CARD || it == DocumentType.PASSPORT);
+                .anyMatch(it -> it == DocumentType.PASSPORT);
+    }
+
+    private static boolean containsSecondaryDocument(final Collection<DocumentVerificationEntity> documentVerifications) {
```

Comment:
I do not like this method name.

A primary document is: ID card, passport.
A secondary document is: driving license.

The method name should be e.g. `containsSecondDocument` or `containsAnotherDocument`, not `secondary`, this word has a special meaning.

Reply:
Renamed to `containsSecondDocument`.